### PR TITLE
If SpinBox text can't be parsed use last updated text

### LINKED ISSUE #88778: Unfocusing empty SpinBox leaves it empty
### Tested versions

4.3 dev3+
4.2.1

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

https://github.com/godotengine/godot/assets/2223172/e4e91811-1486-4f51-9837-b6127205fe97

### Steps to reproduce

1. Go to Grid Snap settings
2. Select a numeric field
3. Remove all text
4. Click another field

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by ryevdokimov**
Fixes: #88778

Before: 

https://github.com/godotengine/godot/assets/105675984/52b27099-a90c-499e-8fee-c01144526fc6

After:

https://github.com/godotengine/godot/assets/105675984/2d2a368c-0a74-470f-92d4-4a84452dd057

### CODE REVIEW

File: scene/gui/spin_box.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ void SpinBox::_text_submitted(const String &p_string) {
 
 		err = expr->parse(text);
 		if (err != OK) {
-			return;
+			expr->parse(last_updated_text);
```
Comment (AThousandShips) [inline scene/gui/spin_box.cpp:86]:
```suggestion
			err = expr->parse(last_updated_text);
		}
		if (err != OK) {
			return;
		}
```
Or things will break

Comment (ryevdokimov) [reply scene/gui/spin_box.cpp:86]: The value is initialized as 0 as part of the range class it inherits, but I agree it's better safe than sorry.

**Comment by KoBeWi** [comment]
Apparently the issue is a regression from #80699
The bug was already fixed in the past like this: #81094 It should be sufficient, so I think reapplying the old fix would be better.

**Comment by ryevdokimov** [comment]
I agree, it does exactly the same thing. Tested and pushed.

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**